### PR TITLE
UICIRCLOG-24 fix non sorting by date by default

### PR DESCRIPTION
**Pull request by axelhunn**
UICIRCLOG-24 Fix non sorting by date by default

## Purpose
Log entries should by default be sorted by date, with most recent ones on top.
Also this should be the second sorting criteria, when user explicitly sorted on any other field.

## Approach
Util function `buildLogEventsQuery()` now includes this logic.
Sorting direction is defined in constants.
Tests are added to check presence of proper `sortBy` clause in generated CQL queries.

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/CirculationLogList/utils.js [hunk 1]
```diff
@@ -2,13 +2,24 @@ import { flatten } from 'lodash';
...
+  SORTING_DIRECTION_PARAMETER,
 } from '@folio/stripes-acq-components';
 
-import { CUSTOM_FILTERS } from './constants';
+import { CUSTOM_FILTERS, DATE_DEFAULT_SORTING_DIRECTION } from './constants';
+
+const queryBuilder = makeQueryBuilder(
+  'cql.allRecords=1',
+  undefined,
+  undefined,
```
Comment (aliaksei-chumakou) [inline src/CirculationLogList/utils.js:14]: shouldn't default sorting be here?

**Review by aliaksei-chumakou — APPROVED** [review]

File: src/CirculationLogList/utils.js (hunk 1 above)
Comment (axelhunn) [reply src/CirculationLogList/utils.js:14]: If we put it here, it should be in plain text like `sortby date/sort.descending` (if I get it right looking through `makeQueryBuilder` code)

Not very convenient, more fragile in my point of view

**Review by uladislausamets — APPROVED** [review]

File: src/CirculationLogList/utils.js (hunk 1 above)
Comment (NikitaSedyx) [reply src/CirculationLogList/utils.js:14]: and what's wrong? it's an argument for default sorting

**Review by AndreiShumski2 — APPROVED** [review]

File: src/CirculationLogList/utils.js (hunk 1 above)
Comment (axelhunn) [reply src/CirculationLogList/utils.js:14]: Yeah, changed my mind: we're doing plain text anyway.

File: src/CirculationLogList/utils.js [hunk 2]
```diff
@@ -23,10 +33,12 @@ export const buildLogEventsQuery = queryParams => {
...
 
-  return makeQueryBuilder(
-    'cql.allRecords=1',
-    undefined,
-    undefined,
-    CUSTOM_FILTERS,
-  )(formattedQueryParams);
+  let query = queryBuilder(formattedQueryParams);
+
+  // We want to sort by date inside any other sorting
```
Comment (NikitaSedyx) [inline src/CirculationLogList/utils.js:38]: is it described in the ticket that we need `to sort by date inside any other sorting` ? basically we need just sort by date by default

Comment (axelhunn) [reply src/CirculationLogList/utils.js:38]: I think the point of the issue is that user wants the descending sorting by date as this is natural for a log.
So if we sort explicitly by another column, it's just natural to keep date sorting this way. 

Commented on this in the ticket: https://issues.folio.org/browse/UICIRCLOG-24?focusedCommentId=90052&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-90052

**Review by NikitaSedyx — APPROVED** [review]

**Pull request merged**